### PR TITLE
added model to project

### PR DESCRIPTION
**Pull request by chrislicodes**
Belongs to #10.

Added model to the project.

### CODE REVIEW

File: js/model.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * deletes a todo from the state of the model
+     * @param todoID {String} - ID of the todo to be deleted *
+     */
+    const newState = [...this.#state.todos];
+    const todoIndex = newState.findIndex((todoObj) => todoObj.id === todoID);
+
+    newState.splice(todoIndex, 1);
+
+    this.#state.todos = newState;
```
Comment (artdaw) [inline js/model.js:29]: No need to use splice and findIndex, just use filter:
```suggestion
this.#state.todos = this.#state.todos.filter((todo) => todo !== todoID);
```

**Review by artdaw — CHANGES_REQUESTED** [review]
Please, apply some changes I suggest. 
Learn about [`Array#filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) and [`Array#map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

File: js/model.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * marks a todo as completed
+     * @param todoID {String} - ID of the todo to be completed*
+     */
+    const newState = [...this.#state.todos];
+    const todoObj = newState.find((todo) => todo.id === todoID);
+
+    todoObj.completeTask();
+
+    this.#state.todos = newState;
```
[the remark this answers is not in the document]
Comment (chrislicodes) [reply js/model.js]: I made changes to the completion methods of the entity for this to work

File: js/model.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * set completed status of a todo to false
+     * @param todoID {String} - ID of the todo to remove the completed status
+     */
+    const newState = [...this.#state.todos];
+    const todoObj = newState.find((todo) => todo.id === todoID);
+
+    todoObj.incompleteTask();
+
+    this.#state.todos = newState;
```
[the remark this answers is not in the document]
Comment (chrislicodes) [reply js/model.js]: I made changes to the completion methods of the entity for this to work

**Pull request merged**